src/libraries/System.Management/src/System/Management/ManagementQuery.cs

```diff
@@ -1040,13 +1040,13 @@ protected internal override void ParseQuery(string query)
                 // Should start with "select"
                 if ((q.Length < keyword.Length) ||
                     (0 != string.Compare(q, 0, keyword, 0, keyword.Length, StringComparison.OrdinalIgnoreCase)))
-                    throw new ArgumentException(SR.InvalidQuery, "select");
```

Comment:
Hmm.  This change and the others like it in this file is actually losing information.  Previously the developer would get an exception like:
```
The Query string supplied was invalid or improperly formed
Param name: select
```
and they'd be more easily able to debug what went wrong.  With this change, the specific aspect of what failed is going to be lost.

If we want to change this to `nameof(query)` (which is fine), I think we should also update the exception message text to have a placeholder, and format the existing parameter name into it, e.g.
```C#
throw new ArgumentException(SR.Format(SR.InvalidQuery, "select"), nameof(query));
```

Reply:
Right, I forgot to update the message

Reply:
Updating with: `SR.Format(SR.InvalidQueryTokenExpected, keyword)` 
Added new string resource `InvalidQueryTokenExpected` which value is `The Query string supplied was invalid or improperly formed. Token `{0}` is expected`